synedrion/src/uint/public_signed.rs

```diff
@@ -183,25 +175,25 @@ impl<T> PublicSigned<T>
 where
     T: Integer + Bounded + Encoding,
 {
-    /// Returns a value in range `[-bound, bound]` derived from an extendable-output hash.
+    /// Returns a value in range `±2^{exp}` derived from an extendable-output hash.
     ///
-    /// This method should be used for deriving non-interactive challenges,
-    /// since it is guaranteed to produce the same results on 32- and 64-bit platforms.
-    pub fn from_xof_reader_bounded(rng: &mut impl XofReader, bound: &NonZero<T>) -> Self {
-        let bound_bits = bound.as_ref().bits_vartime();
-        assert!(bound_bits < <T as Bounded>::BITS);
-        // Will not overflow because of the assertion above
-        let positive_bound = bound
-            .as_ref()
-            .overflowing_shl_vartime(1)
-            .expect("Just asserted that bound is smaller than precision; qed")
-            .checked_add(&T::one())
-            .expect("does not overflow since we're adding 1 to an even number");
-        let positive_result = uint_from_xof(
-            rng,
-            &NonZero::new(positive_bound).expect("Guaranteed to be greater than zero because we added 1"),
+    /// Note that in the paper's definitions, `±2^{exp}` is equivalent to `[-2^(x-1)+1, 2^(x-1)]`
+    /// (see Section 3, Groups & Fields).
+    pub fn from_xof_reader_in_exponent_range(reader: &mut impl XofReader, exp: u32) -> Self {
+        assert!(exp > 0, "`exp` must be greater than zero");
+        assert!(
+            exp < T::BITS,
+            "Out of bounds: `exp` was {exp} but must be smaller or equal to {}",
+            T::BITS
         );
-        Self::new_from_unsigned(positive_result.wrapping_sub(bound.as_ref()), bound_bits)
+
+        let positive_result = T::from_xof_reader(reader, exp);
+        let shift = T::one()
+            .overflowing_shl_vartime(exp - 1)
+            .expect("does not overflow because of the assertions above")
+            .checked_sub(&T::one())
+            .expect("does not overflow because of the assertions above");
+        Self::new_from_unsigned(positive_result.wrapping_sub(&shift), exp)
```

Comment:
`new_from_unsigned` clones; do you think we can use `new_from_abs` here?

Reply:
`new_from_abs` would only allows us to generate numbers in a symmetric range, the current approach produces the required `[-2^(x-1)+1, 2^(x-1)]`. Granted, that +1 is negligible for our typical values of `x`, but so are the clones. Perhaps it will be easier to make `new_from_unsigned()` not clone - we only clone to calculate its bound anyway.